src/snowflake/connector/util_text.py

```diff
@@ -7,223 +7,86 @@
 import logging
 import re
 from io import StringIO
-from typing import Optional
 
-COMMENT_PATTERN_RE = re.compile(r"^\s*\-\-")
-EMPTY_LINE_RE = re.compile(r"^\s*$")
+from .antlr.sql_separator import sep_sqls
 
 _logger = logging.getLogger(__name__)
 
+# the pattern match a full line of comment
+sf_comment_cmd_pattern = re.compile(r"(\n|\r)+\s*--sf:.+(\n|\r)+")
 
-class SQLDelimiter(object):
-    """Class that wraps a SQL delimiter string.
 
-    Since split_statements is a generator this mutable object will allow it change while executing.
-    """
-
-    def __str__(self):
-        return self.sql_delimiter
-
-    def __init__(self, sql_delimiter: str = ";"):
-        """Initializes SQLDelimiter with a string."""
-        self.sql_delimiter = sql_delimiter
+def is_put_or_get(stmt):
+    return stmt.strip()[:3].upper() in ("PUT", "GET")
 
 
 def split_statements(
     buf: StringIO,
-    remove_comments: bool = False,
```

Comment:
While you fixed the tests to expect the comments, doesn't SnowSQL expect them to be removed?

The comments will now show up in customers' query history too. THis might need to be announced too.